cl/alerts/views.py

```diff
@@ -177,32 +177,96 @@ def subscribe_docket_alert(
     request: HttpRequest, secret_key: str
 ) -> HttpResponse:
     """Subscribe a user to a docket alert based on the alert secret_key."""
-    docket_alert = flip_docket_alert(secret_key, DocketAlert.SUBSCRIPTION)
-    return render(
-        request,
-        "docket_alert.html",
-        {"docket_alert": docket_alert, "private": True, "enabled": True},
+    response = docket_alert_toggle_confirmation(
+        request, secret_key, DocketAlert.SUBSCRIPTION
     )
+    return response
 
 
 @ratelimit_deny_list
 def unsubscribe_docket_alert(
     request: HttpRequest, secret_key: str
 ) -> HttpResponse:
     """Unsubscribe a user from a docket alert based on the alert secret_key."""
-    docket_alert = flip_docket_alert(secret_key, DocketAlert.UNSUBSCRIPTION)
-    # Send Unsubscription confirmation email to the user
-    send_unsubscription_confirmation.delay(docket_alert.pk)
-    return render(
-        request,
-        "docket_alert.html",
-        {"docket_alert": docket_alert, "private": True, "enabled": False},
+
+    response = docket_alert_toggle_confirmation(
+        request, secret_key, DocketAlert.UNSUBSCRIPTION
     )
+    return response
 
 
 def flip_docket_alert(secret_key: str, alert_type: int) -> DocketAlert:
     """Flip the alert_type for a docket alert."""
     docket_alert = get_object_or_404(DocketAlert, secret_key=secret_key)
     docket_alert.alert_type = alert_type
     docket_alert.save()
+    if alert_type == DocketAlert.UNSUBSCRIPTION:
+        # Send Unsubscription confirmation email to the user
+        send_unsubscription_confirmation.delay(docket_alert.pk)
     return docket_alert
+
+
+def docket_alert_toggle_confirmation(
+    request: HttpRequest, secret_key: str, alert_type: int
+) -> HttpResponse:
+    """Show a confirmation or success page for toggling docket alerts."""
+
+    enabled = False
+    if alert_type == DocketAlert.SUBSCRIPTION:
+        enabled = True
+    docket_alert = DocketAlert.objects.get(secret_key=secret_key)
+    # Handle confirmation form POST requests
+    if request.method == "POST":
+        form = DocketAlertConfirmForm(request.POST)
+        if form.is_valid():
+            flip_docket_alert(secret_key, alert_type)
```

Comment:
Unless `flip_docket_alert` is being used elsewhere, why send it the `secret_key` instead of the alert object? Doing so costs a query and complicates the code.

Reply:
Yeah, this is better, changed. Also, I added a condition to only flip the `alert_type` if it's different from the new `alert_type`. For example, if a user goes to an old email and unsubscribes again it won't trigger the unsubscription email again.